### PR TITLE
Migrate new discord js with minor rework

### PR DESCRIPTION
**Pull request by ChristopherVR**
- [x] Refactored files
- [x]  Restructured project
- [x] Update NPM packages
- [x] Migrate to latest DiscordJS logic
- [x] Improved stability
- [x] Improved logging messages
- [x] Added stub logger (to be replaced in future with a proper logger)
- [x] Added logic to prevent NodeJS from crashing when an exception is thrown
- [x] Removed magic words (replaced with Enums)
- [x] Reduced default volume to 65%
- [x] Resolved memory leak with multiple Extractors being registered to the discord player
- [x]  Improved handling of translations
- [x] Added a few more specific error messages
- [x] Converted project from CommonJS to a ESM module
- [x] Replaced all `requires` with `import/export`
- [x] Added `deferReply` to prevent timeout issues on play commands
- [x] Convert eslintrc.json to eslint flat config (all other config styles will be deprecated in V9)
- [x] Update eslint rules to the new standards
- [x] Write JSDocs for functions
- [x] Replace stub logger with a proper logger
- [x] Deploy the project using Vercel
- [x] Resolve issue with duplicate commands showing in discord
- [x] Ensure the application still works through docker container
- [ ] Update project to add report metrics to Vercel
- [ ] Update project to add logging supported by Vercel
- [ ] Update Readme.md with more detailed information on how to setup the project
- [x] Test each command to ensure they work

### CODE REVIEW

File: src/server/setup.ts [hunk 1]
```diff
@@ -32,12 +35,10 @@
     }),
   );
 
-  server.get('/', (_req, res) => {
-    res.sendFile(path.join(__dirname, '/index.html'));
-  });
+  server.get('/', (_, res) => res.sendFile(path.join(path.dirname('/index.html'))));
```
[the remark this answers is not in the document]
Comment (ElHefe3) [reply src/server/setup.ts:38]: We can leave this, but we'll have to change it before deploying to a public server.

File: src/bot/interactionCreate.ts [hunk 2]
```diff
@@ -1,24 +0,0 @@
-import { ChatInputCommandInteraction, Client, Interaction } from 'discord.js';
-import { getCommands } from '../commands';
-
```
Comment (ElHefe3) [inline src/bot/interactionCreate.ts:3]: Separate external & internal imports

**Review by ElHefe3 — CHANGES_REQUESTED** [review]
Just fix the test

File: src/server/setup.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChristopherVR) [reply src/server/setup.ts:38]: Rate limit is being send on the line before this one.

**Comment by ChristopherVR** [comment]
Manually merged.

**Pull request closed without merging**